### ISSUE TITLE
Computed Property Based on Nested Fragment

### ISSUE THREAD

**Issue by lmcardle** [lytics/ember-data-model-fragments#178]
I have a fragment that looks like this:

```
level1: {
  level2: [
    {
      level3: [
        {
          propFoo: 'bar'
        }
      ]
    }
  ]
}
```

Within a template, if I loop over level's 2 and 3 together, I get live updating (i.e. it works as expected):
```
{{#each level1.level2 as |x|}}
  {{#each x.level3 as |y|}}
    <p>{{y.propFoo}}</p>
  {{/each}}
{{/each}}
```

If I have an action and remove one of the y's, the list displayed is updated with one less y.

I need to move this to the controller in a computed property. However, I can't seem to get the computed property to re-fire when one of the y's gets removed. This is what I have currently:

```
sortedLevel3: Ember.computed('model.level1.level2.@each.level3.[]', function() {
  // do something
  // return sorted level 3 list
}

{{#each sortedLevel3 as |y|}}
  <p>{{y.propFoo}}</p>
{{/each}}
```
While the above works the first time the page is loaded, the sortedLevel3 function never gets called a second time once one of the items in level3's array is removed. Any idea what I am missing?

**Comment by lmcardle** [comment]
One additional piece of info: 
level1 is a regular ember data model
level2 is a fragmentArray
level3 is a fragmentArray

**Comment by workmanw** [comment]
Hey @lmcardle what version of ember-data are you on? Since we're using Ember.ArrayProxy, I suspect this is related to: https://github.com/emberjs/ember.js/issues/12475 . 

If I understand that issue properly, you should be able to use `level2.content.@each` instead (see below). Maybe give that a shot?

```js
sortedLevel3: Ember.computed('model.level1.level2.content.@each.level3.[]', function() {
  // do something
  // return sorted level 3 list
}
```

**Comment by slindberg** [comment]
I believe @workmanw is correct that this is the same issue as emberjs/ember.js#12475. The fragment array base class does the same thing as `RecordArray` by [overriding the `content` property](https://github.com/lytics/ember-data-model-fragments/blob/v2.1.1/packages/model-fragments/lib/fragments/array/stateful.js#L45-L47).

Closing since this is a bug in Ember. The commenters on that issue reported success with adding the `content` property to the dependent key chain, so I would start there if you can't upgrade Ember. It is fixed in Ember 2.3.x, but [it doesn't look like it will be backported to 2.0-2.2](https://github.com/emberjs/ember.js/issues/12475#issuecomment-181442485).
